pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,490 @@
+        )
+
+    def check_errors(self):
+        """
+        Method to read the error status register
+
+        :return error_status: one byte with the error status register content
+        :rtype error_status: int
+        """
+
+        if self.error_status != 0:
+            log.error("HP3478A error detected: $s", self.ERRORS[self.error_status])
+        return self.error_status
+
+    error_status = Instrument.measurement(
+        "E",
+        """Checks the error status register
+        """,
+        cast=int,
+        )
+
+    def display_reset(self):
+        """ Reset the display of the instrument.
+        """
+        self.write("D1")
+
+    display_text = Instrument.setting(
+        "D2%s",
+        """Displays up to 12 upper-case ASCII characters on the display.
+        """,
+        set_process = (lambda x: str.upper(x[0:12])),
+        )
+
+    display_text_no_symbol = Instrument.setting(
+        "D3%s",
+        """Displays up to 12 upper-case ASCII characters on the display and disables all symbols on the display.
+        """,
+        set_process = (lambda x: str.upper(x[0:12])),
+        )
+
+    measure_ACI = Instrument.measurement(
+        MODES["ACI"],
+        """Return the measured value for AC current
+        """,
+        )
+
+    measure_ACV = Instrument.measurement(
+        MODES["ACV"],
+        """Return the measured value for AC Voltage
+        """,
+        )
+
+    measure_DCI = Instrument.measurement(
+        MODES["DCI"],
+        """Return the measured value for DC current
+        """,
+        )
+
+    measure_DCV = Instrument.measurement(
+        MODES["DCV"],
+        """Return the measured value for DC Voltage
+        """,
+        )
+
+    measure_R2W = Instrument.measurement(
+        MODES["R2W"],
+        """Return the measured value for 2-wire resistance
+        """,
+        )
+
+    measure_R4W = Instrument.measurement(
+        MODES["R4W"],
+        """Return the measured value for 4-wire resistance
+        """,
+        )
+
+    measure_Rext = Instrument.measurement(
+        MODES["Rext"],
+        """Return the measured value for extended resistance mode (>30M, 2-wire) resistance
+        """,
+        )
+
+    mode = Instrument.control(
+        "B",
+        "%s",
+        """Return current selected measurment mode, this propery can be set.
+        Allowed values are ACI,ACV,DCI,DCV,R2W,R4W,Rext
+        """,
+        get_process = (lambda x: HP3478A.decode_mode(x)),
+        set_process = (lambda x: HP3478A.MODES[x]),
+        validator = strict_discrete_set,
+        values = MODES,
+        )
+
+    range = Instrument.measurement(
+        "B",
+        """Returns the current measurment range, this property can be set.
+        Valid settings are 3*powers of ten (e.g 0.3,3,30)"
+        for all valid ranges look at HP3478A.RANGES structure
+        """,
+        get_process = lambda x: HP3478A.decode_range(x),
+        )
+
+    @range.setter
+    def range(self,value):
```

Comment:
For reasons of symmetry, while the current arrangement of a measurement + separate setter works, it looks a bit weird - a classic property instead of the measurement would feel more natural to me.